src/components/Sidebar.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useState } from "react";
+import { trpc } from "../utils/trpc";
+import InfiniteScroll from "react-infinite-scroll-component";
+import Spinner from "./Spinner";
+import { Role, Status, User } from "@prisma/client";
+import mapboxgl, { Marker } from "mapbox-gl";
+import { min } from "lodash";
+
+type ScrollableList = {
+  items: User[];
+  idx: number;
+};
+
+const previousMarkers: mapboxgl.Marker[] = [];
```

Comment:
Ok I wrote this code but in retrospect I wonder if it would benefit from being a hook inside the Sidebar component.